diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-07-08 16:00:20.000000000 +0800
+++ b/CMakeLists.txt	2016-07-15 11:13:47.910004939 +0800
@@ -45,8 +45,7 @@
                   gio-2.0>=2.36
                   gio-unix-2.0>=2.36
                   gudev-1.0>=204
-                  libnotify>=0.7.6
-                  url-dispatcher-1>=1)
+                  libnotify>=0.7.6)
 
 include_directories (SYSTEM ${SERVICE_DEPS_INCLUDE_DIRS})
 
diff -Nur a/src/notifier.c b/src/notifier.c
--- a/src/notifier.c	2016-07-08 16:00:20.000000000 +0800
+++ b/src/notifier.c	2016-07-15 11:23:11.452688260 +0800
@@ -18,13 +18,10 @@
  */
 
 #include "datafiles.h"
-#include "dbus-accounts-sound.h"
 #include "dbus-battery.h"
 #include "dbus-shared.h"
 #include "notifier.h"
 
-#include <url-dispatcher.h>
-
 #include <libnotify/notify.h>
 
 #include <glib/gi18n.h>
@@ -81,8 +78,6 @@
   gboolean actions_supported;
 
   GCancellable * cancellable;
-  DbusAccountsServiceSound * accounts_service_sound_proxy;
-  gboolean accounts_service_sound_proxy_pending;
 }
 IndicatorPowerNotifierPrivate;
 
@@ -137,54 +132,6 @@
 }
 
 /***
-****  Sounds
-***/
-
-static void
-on_sound_proxy_ready (GObject      * source_object G_GNUC_UNUSED,
-                      GAsyncResult * res,
-                      gpointer       gself)
-{
-  GError * error;
-  DbusAccountsServiceSound * proxy;
-
-  error = NULL;
-  proxy = dbus_accounts_service_sound_proxy_new_for_bus_finish (res, &error);
-  if (error != NULL)
-    {
-      if (!g_error_matches(error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
-        {
-          get_priv(gself)->accounts_service_sound_proxy_pending = FALSE;
-          g_debug("%s Couldn't find accounts service sound proxy: %s", G_STRLOC, error->message);
-        }
-
-      g_clear_error(&error);
-    }
-  else
-    {
-      IndicatorPowerNotifier * const self = INDICATOR_POWER_NOTIFIER(gself);
-      priv_t * const p = get_priv (self);
-      g_clear_object (&p->accounts_service_sound_proxy);
-      p->accounts_service_sound_proxy = proxy;
-      p->accounts_service_sound_proxy_pending = FALSE;
-    }
-}
-
-static gboolean
-silent_mode (IndicatorPowerNotifier * self)
-{
-  priv_t * const p = get_priv (self);
-
-  /* if we don't have a proxy yet, assume we're in silent mode
-     as a "do no harm" level of response */
-  if (p->accounts_service_sound_proxy_pending)
-    return TRUE;
-
-  return (p->accounts_service_sound_proxy != NULL)
-      && dbus_accounts_service_sound_get_silent_mode(p->accounts_service_sound_proxy);
-}
-
-/***
 ****  Notifications
 ***/
 
@@ -226,7 +173,7 @@
                             char               * action    G_GNUC_UNUSED,
                             gpointer             user_data G_GNUC_UNUSED)
 {
-  url_dispatch_send("settings:///system/battery", NULL, NULL);
+
 }
 
 static void
@@ -298,21 +245,6 @@
 
   if (are_actions_supported(self))
     {
-      if (!silent_mode(self))
-        {
-          gchar* filename = datafile_find(DATAFILE_TYPE_SOUND, LOW_BATTERY_SOUND);
-          if (filename != NULL)
-            {
-              gchar * uri = g_filename_to_uri(filename, NULL, NULL);
-              notify_notification_set_hint(nn, "sound-file", g_variant_new_take_string(uri));
-              g_clear_pointer(&filename, g_free);
-            }
-          else
-            {
-              g_warning("Unable to find '%s' in XDG data dirs", LOW_BATTERY_SOUND);
-            }
-        }
-
       notify_notification_set_hint(nn, "x-canonical-snap-decisions", g_variant_new_string("true"));
       notify_notification_set_hint(nn, "x-canonical-non-shaped-icon", g_variant_new_string("true"));
       notify_notification_set_hint(nn, "x-canonical-private-affirmative-tint", g_variant_new_string("true"));
@@ -439,7 +371,6 @@
   notification_clear (self);
   indicator_power_notifier_set_battery (self, NULL);
   g_clear_object (&p->dbus_battery);
-  g_clear_object (&p->accounts_service_sound_proxy);
 
   G_OBJECT_CLASS (indicator_power_notifier_parent_class)->dispose (o);
 }
@@ -473,18 +404,6 @@
 
   if (!instance_count++ && !notify_init(SERVICE_EXEC))
     g_critical("Unable to initialize libnotify! Notifications might not be shown.");
-
-  p->accounts_service_sound_proxy_pending = TRUE;
-  gchar* object_path = g_strdup_printf("/org/freedesktop/Accounts/User%lu", (gulong)getuid());
-  dbus_accounts_service_sound_proxy_new_for_bus(
-    G_BUS_TYPE_SYSTEM,
-    G_DBUS_PROXY_FLAGS_GET_INVALIDATED_PROPERTIES,
-    "org.freedesktop.Accounts",
-    object_path,
-    p->cancellable,
-    on_sound_proxy_ready,
-    self);
-  g_clear_pointer(&object_path, g_free);
 }
 
 static void
diff -Nur a/src/service.c b/src/service.c
--- a/src/service.c	2016-07-08 16:00:29.000000000 +0800
+++ b/src/service.c	2016-07-15 11:12:07.724834388 +0800
@@ -20,7 +20,6 @@
 
 #include <glib/gi18n.h>
 #include <gio/gio.h>
-#include <url-dispatcher.h>
 
 #include "brightness.h"
 #include "dbus-shared.h"
@@ -802,12 +801,7 @@
 
   if (control_center_cmd == NULL)
     {
-      if (g_getenv ("MIR_SOCKET") != NULL)
-        {
-          url_dispatch_send("settings:///system/battery", NULL, NULL);
-          return;
-        }
-      else if (!g_strcmp0 (g_getenv ("DESKTOP_SESSION"), "xubuntu"))
+      if (!g_strcmp0 (g_getenv ("DESKTOP_SESSION"), "xubuntu"))
         {
           control_center_cmd = "xfce4-power-manager-settings";
         }
@@ -852,7 +846,7 @@
                              GVariant      * param  G_GNUC_UNUSED,
                              gpointer        gself  G_GNUC_UNUSED)
 {
-  url_dispatch_send("settings:///system/battery", NULL, NULL);
+  
 }
 
 /***
