diff -Nur a/services/CMakeLists.txt b/services/CMakeLists.txt
--- a/services/CMakeLists.txt	2016-04-15 13:32:34.000000000 +0800
+++ b/services/CMakeLists.txt	2016-04-17 20:59:56.078586352 +0800
@@ -12,7 +12,6 @@
     indicator3-0.4>=12.10.2
     x11
     libido3-0.1>=13.0.0
-    libupstart
 )
 
 pkg_check_modules(SERVICE_DEPS REQUIRED ${UNITY_PANEL_SERVICE_DEPS})
@@ -52,9 +51,3 @@
 add_executable(unity-panel-service ${PANEL_SOURCES})
 target_link_libraries(unity-panel-service ${LIBS})
 install(TARGETS unity-panel-service DESTINATION ${CMAKE_INSTALL_LIBDIR}/unity/)
-
-configure_file(unity-panel-service.conf.in ${CMAKE_CURRENT_BINARY_DIR}/unity-panel-service.conf)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/unity-panel-service.conf DESTINATION ${CMAKE_INSTALL_DATADIR}/upstart/sessions)
-
-configure_file(unity-panel-service-lockscreen.conf.in ${CMAKE_CURRENT_BINARY_DIR}/unity-panel-service-lockscreen.conf)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/unity-panel-service-lockscreen.conf DESTINATION ${CMAKE_INSTALL_DATADIR}/upstart/sessions)
diff -Nur a/services/panel-service.c b/services/panel-service.c
--- a/services/panel-service.c	2016-04-15 13:32:34.000000000 +0800
+++ b/services/panel-service.c	2016-04-17 21:01:32.075899348 +0800
@@ -34,8 +34,6 @@
 #include <X11/XF86keysym.h>
 #include <X11/extensions/XInput2.h>
 
-#include <upstart/upstart-dbus.h>
-
 G_DEFINE_TYPE (PanelService, panel_service, G_TYPE_OBJECT);
 
 #define GET_PRIVATE(o) \
@@ -804,6 +802,9 @@
 static void
 emit_upstart_event (const gchar *event)
 {
+ #if 1
+  (void) event;
+ #else
   const gchar *upstartsession = g_getenv ("UPSTART_SESSION");
 
   if (!upstartsession)
@@ -840,6 +841,7 @@
     }
 
   g_object_unref (conn);
+#endif
 }
 
 
diff -Nur a/tests/CMakeLists.txt b/tests/CMakeLists.txt
--- a/tests/CMakeLists.txt	2016-04-15 13:32:34.000000000 +0800
+++ b/tests/CMakeLists.txt	2016-04-17 21:10:53.737420688 +0800
@@ -8,7 +8,7 @@
 #
 # Unit tests
 #
-set (TEST_DEPS "${UNITY_PLUGIN_DEPS};unity>=4.99.0;libupstart;")
+set (TEST_DEPS "${UNITY_PLUGIN_DEPS};unity>=4.99.0;")
 pkg_check_modules (TEST_UNIT_DEPS REQUIRED ${TEST_DEPS})
 
 string (REPLACE ";" " " TEST_UNIT_DEPS_CFLAGS_OTHER "${TEST_UNIT_CFLAGS_OTHER}")
diff -Nur a/tools/unity.cmake b/tools/unity.cmake
--- a/tools/unity.cmake	2016-04-15 13:32:57.000000000 +0800
+++ b/tools/unity.cmake	2016-04-17 21:07:24.517691769 +0800
@@ -138,9 +138,9 @@
 
     try:
         options.debug_mode = 2 if options.advanced_debug else 1 if options.debug else 0
-        if is_upstart_session(): subprocess.call(["stop", "unity-panel-service"])
+        subprocess.call(["killall", "unity-panel-service"])
         unity_instance = process_and_start_unity (compiz_args)
-        if is_upstart_session(): subprocess.call(["start", "unity-panel-service"])
+        subprocess.call(["/lib/ld-linux-x86-64.so.2", "/usr/lib/unity/unity-panel-service"])
         unity_instance.wait()
     except KeyboardInterrupt as e:
         try:
diff -Nur a/unity-shared/UpstartWrapper.cpp b/unity-shared/UpstartWrapper.cpp
--- a/unity-shared/UpstartWrapper.cpp	2016-04-15 13:32:34.000000000 +0800
+++ b/unity-shared/UpstartWrapper.cpp	2016-04-17 21:35:13.150300198 +0800
@@ -20,7 +20,6 @@
 #include "UpstartWrapper.h"
 
 #include <UnityCore/GLibDBusProxy.h>
-#include <upstart/upstart-dbus.h>
 
 namespace unity
 {
@@ -36,19 +35,30 @@
 
   void Emit(std::string const& name);
 
+#if 0
 private:
   glib::DBusProxy::Ptr upstart_proxy_;
+#endif
 };
 
+
 UpstartWrapper::Impl::Impl(bool test_mode)
 {
+#if 1
+  (void) test_mode;
+#else
   upstart_proxy_ = std::make_shared<unity::glib::DBusProxy>(test_mode ?  "com.canonical.Unity.Test.Upstart" : DBUS_SERVICE_UPSTART,
                                                             DBUS_PATH_UPSTART, DBUS_INTERFACE_UPSTART);
+#endif
 }
 
 void UpstartWrapper::Impl::Emit(std::string const& name)
 {
+#if 1
+  (void) name;
+#else
   upstart_proxy_->Call("EmitEvent", g_variant_new("(sasb)", name.c_str(), nullptr, 0));
+#endif
 }
 
 //
@@ -68,7 +78,11 @@
 
 void UpstartWrapper::Emit(std::string const& name)
 {
+#if 1
+  (void) name;
+#else
   pimpl_->Emit(name);
+#endif
 }
 
 }
