diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-04-12 19:58:22.000000000 +0800
+++ b/CMakeLists.txt	2016-04-17 15:10:48.990655081 +0800
@@ -29,8 +29,7 @@
 include (FindPkgConfig)
 pkg_check_modules (SERVICE REQUIRED
                    glib-2.0>=2.36
-                   gio-unix-2.0>=2.36
-                   url-dispatcher-1>=1)
+                   gio-unix-2.0>=2.36)
 include_directories (${SERVICE_INCLUDE_DIRS})
 
 set (CC_WARNING_ARGS " -Wall -pedantic -Wextra -Wno-missing-field-initializers")
diff -Nur a/src/backend-dbus/actions.c b/src/backend-dbus/actions.c
--- a/src/backend-dbus/actions.c	2016-04-12 19:58:28.000000000 +0800
+++ b/src/backend-dbus/actions.c	2016-04-17 15:11:42.645348048 +0800
@@ -19,7 +19,6 @@
 
 #include <glib.h>
 #include <glib/gi18n.h>
-#include <url-dispatcher.h>
 
 #include "dbus-end-session-dialog.h"
 #include "dbus-login1-manager.h"
@@ -831,9 +830,6 @@
 static void
 my_help (IndicatorSessionActions * self G_GNUC_UNUSED)
 {
-  if (g_getenv ("MIR_SOCKET") != NULL)
-    url_dispatch_send("http://www.askubuntu.com", NULL, NULL);
-  else
     run_outside_app ("yelp");
 }
 
@@ -872,9 +868,7 @@
 static void
 my_settings (IndicatorSessionActions * self G_GNUC_UNUSED)
 {
-  if (g_getenv ("MIR_SOCKET") != NULL)
-    url_dispatch_send("settings:///system", NULL, NULL);
-  else if (have_unity_control_center ())
+  if (have_unity_control_center ())
     run_outside_app ("unity-control-center");
   else
     run_outside_app ("gnome-control-center");
@@ -883,9 +877,7 @@
 static void
 my_online_accounts (IndicatorSessionActions * self G_GNUC_UNUSED)
 {
-  if (g_getenv ("MIR_SOCKET") != NULL)
-    url_dispatch_send("settings:///system/online-accounts", NULL, NULL);
-  else if (have_unity_control_center ())
+  if (have_unity_control_center ())
     run_outside_app ("unity-control-center credentials");
   else
     run_outside_app ("gnome-control-center credentials");
@@ -894,9 +886,7 @@
 static void
 my_about (IndicatorSessionActions * self G_GNUC_UNUSED)
 {
-  if (g_getenv ("MIR_SOCKET") != NULL)
-    url_dispatch_send("settings:///system/about", NULL, NULL);
-  else if (have_unity_control_center ())
+  if (have_unity_control_center ())
     run_outside_app ("unity-control-center info");
   else
     run_outside_app ("gnome-control-center info");
